packages/yaml-editor/src/hooks/useDecoration.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { RefObject, useEffect, useMemo, useState } from 'react'
+import * as monaco from 'monaco-editor'
+import { getOutlineModel, processOutlineModel } from '../utils/outline-model-to-path'
+
+type UseDecoration = (arg: {
+  editorRef?: RefObject<monaco.editor.IStandaloneCodeEditor>
+  selection?: {
+    path: string
+    className: string
+    revealInCenter?: boolean
+  }
+}) => void
+
+export const useDecoration: UseDecoration = ({ editorRef, selection }) => {
+  const [range, setRange] = useState<monaco.IRange>()
+
+  const computeRange = useMemo(
+    () =>
+      async (_editorRef?: RefObject<monaco.editor.IStandaloneCodeEditor>, path?: string, revealInCenter?: boolean) => {
+        const model = editorRef?.current?.getModel()
+
+        if (!model || !path) {
+          return setRange(undefined)
+        }
+
+        const outlineModel = await getOutlineModel(model)
+        const pathSymbolMap = processOutlineModel(outlineModel)
+        const range = pathSymbolMap.get(path)?.range
+
+        if (range) {
+          setRange(range)
+
+          if (revealInCenter) {
+            editorRef?.current?.revealLineInCenter(range.startLineNumber)
+          }
+        }
+      },
+    []
+  )
+
+  useEffect(() => {
+    const timeout = setTimeout(() => computeRange(editorRef, selection?.path, selection?.revealInCenter), 100)
```

Comment:
what's the purpose of the 100ms hardcoded timeout?
can we make it configurable via props if a delay is required?
or maybe replace this with a `requestIdleCallback()` if the intent is to just queue the action?

Reply:
Will be fixed in the following PR.